modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/twostep/GridReduceQueryExecutor.java

```diff
@@ -466,11 +469,9 @@ else if (msg.page() == 0) // Count down only on each first page received.
                 partsMap = nodesParts.partitionsMap();
                 qryMap = nodesParts.queryPartitionsMap();
 
-                if (nodes == null)
+                if (F.isEmpty(nodes))
```

Comment:
Seems that similar condition `(nodes == null)` , where `nodes` comes from `nodesForPartitions.nodes()` is used in the `update()` method here. Should we replace it with `F.isEmpty()` too (and add corresponding test)?